### PR TITLE
Add time series related information to get data stream API

### LINKED ISSUE #83518: Include time series temporal ranges to get data stream api for data streams in time series index mode
Change the get data stream api to include the following properties of a data stream:

- ~~Add `index_mode` from `DataStream` to the response.~~
- For each backing index of a data stream with time series index mode, include the temporal range the backing index represents (based on the `index.time_series.start_time` and `index.time_series.end_time` index settings.

### PR DESCRIPTION
**Pull request by martijnvg**
In case if a data stream is a time series data stream then include time series information.
This includes the continuous temporal ranges a time series data stream encapsulates.
This is computed based on combing the index.time_series.start_time and index.time_series.end_time
ranges of all backing indices of a time series data stream

Closes #83518

### CODE REVIEW

**Comment by martijnvg** [comment]
This change add `time_series` snippet to get data streams api response if a data stream is a time series data stream.
Example `GET /_data_stream` run:

```
{
    "data_streams": [
        {
            "name": "k8s",
            "timestamp_field": {
                "name": "@timestamp"
            },
            "indices": [
                {
                    "index_name": ".ds-k8s-2022.05.04-000001",
                    "index_uuid": "1C1g7at-RW6vz0kHEGx1ZA"
                }
            ],
            "generation": 1,
            "status": "GREEN",
            "template": "k8s-template",
            "ilm_policy": "7-days-1gb-rollover",
            "hidden": false,
            "system": false,
            "allow_custom_routing": false,
            "replicated": false,
            "time_series": {
                "temporal_ranges": [
                    {
                        "start": "2022-05-04T12:14:53.000Z",
                        "end": "2022-05-04T12:24:53.000Z"
                    }
                ]
            }
        }
    ]
}
```

**Review by sethmlarson — APPROVED** [review]
LGTM from an API perspective

**Comment by martijnvg** [comment]
> LGTM from an API perspective

Thanks @sethmlarson for reviewing!

File: modules/data-streams/src/main/java/org/elasticsearch/datastreams/action/GetDataStreamsTransportAction.java [hunk 1]
```diff
@@ -105,11 +119,41 @@ protected void masterOperation(
                 state,
                 dataStream.getIndices().stream().map(Index::getName).toArray(String[]::new)
             );
+
+            GetDataStreamAction.Response.TimeSeries timeSeries = null;
+            if (dataStream.getIndexMode() == IndexMode.TIME_SERIES) {
+                List<Tuple<Instant, Instant>> ranges = new ArrayList<>();
+                Tuple<Instant, Instant> current = null;
+                for (Index index : dataStream.getIndices()) {
```
[the remark this answers is not in the document]
Comment (martijnvg) [reply modules/data-streams/src/main/java/org/elasticsearch/datastreams/action/GetDataStreamsTransportAction.java:127]: The indices are in order in which they are created. (new data stream, rollover, rollover again -> 3 backing indices that are in order). So the start / end times should be in order.

I will add an assertion and ensure a warning log gets printed (if this ever should occur in production). Which I don't think should be the case.

**Review by nik9000 — APPROVED** [review]

**Comment by martijnvg** [comment]
@elasticmachine update branch

**Pull request merged**